elasticgraph-graphql/spec/unit/elastic_graph/graphql/datastore_query/search_index_expression_spec.rb

```diff
@@ -356,6 +356,59 @@ class GraphQL
             end
           end
 
+          context "when `all_of` is used" do
+            it "ANDs together the date filter criteria from the `all_of` subfilters when determining what indices to exclude" do
+              parts1 = search_index_expression_parts_for({"all_of" => [
+                {"created_at" => {"lt" => "2021-03-01T00:00:00Z"}},
+                {"created_at" => {"gt" => "2021-01-15T12:30:00Z"}}
+              ]})
+
+              # Order of subfilters shouldn't matter
+              parts2 = search_index_expression_parts_for({"all_of" => [
+                {"created_at" => {"gt" => "2021-01-15T12:30:00Z"}},
+                {"created_at" => {"lt" => "2021-03-01T00:00:00Z"}}
+              ]})
+
+              # It shouldn't matter if the all_of is nested or on the outside
+              parts3 = search_index_expression_parts_for({"created_at" => {"all_of" => [
+                {"gt" => "2021-01-15T12:30:00Z"},
+                {"lt" => "2021-03-01T00:00:00Z"}
+              ]}})
+
+              expect(parts1).to eq(parts2).and eq(parts3).and target_widget_indices_excluding_2021_months("03", "04", "05", "06", "07", "08", "09", "10", "11", "12")
+            end
+
+            it "excludes all indices when one of the `all_of` subfilters excludes all indices" do
+              parts = search_index_expression_parts_for({"all_of" => [
+                {"created_at" => {"lt" => "2021-03-01T00:00:00Z"}},
+                {"created_at" => {"gt" => "2021-08-15T12:30:00Z"}}
+              ]})
+
+              expect(parts).to target_no_indices
+            end
+
+            it "excludes no indices when we have an `all_of: []` filter because that will match all results" do
+              parts = search_index_expression_parts_for({"all_of" => []})
+
+              expect(parts).to target_all_widget_indices
+            end
+
+            it "excludes no indices when we have an `all_of: [{field: nil}]` filter because that will match all results" do
+              parts = search_index_expression_parts_for({"all_of" => [{"created_at" => nil}]})
+
+              expect(parts).to target_all_widget_indices
+            end
+
+            it "excludes no indices when we have an `all_of: [{field: nil}, {...}]` filter because first clause matches all results" do
```

Comment:
This test description contradicts what the test shows happens. (I think what the test shows happens is correct.)